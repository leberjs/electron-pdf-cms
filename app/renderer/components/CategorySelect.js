import React, { Component } from 'react'
import Autosuggest from 'react-autosuggest'
import TextField from 'material-ui/TextField'
import Paper from 'material-ui/Paper'
import { MenuItem } from 'material-ui/Menu'
import match from 'autosuggest-highlight/match'
import parse from 'autosuggest-highlight/parse'
import { withStyles } from 'material-ui/styles'

const styles = (theme) => ({
  container: {
    flexGrow: 1,
    position: 'relative'
  },
  suggestionsContainerOpen: {
    position: 'absolute',
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit * 3,
    left: 0,
    right: 0
  },
  suggestion: {
    display: 'block'
  },
  suggestionsList: {
    margin: 0,
    padding: 0,
    listStyleType: 'none'
  },
  textField: {
    minWidth: 300,
    marginBottom: 15
  },
})

/** renders */

function renderInput(inputProps) {
  const { classes, autoFocus, value, ref, ...other } = inputProps

  return (
    <TextField
      autoFocus={autoFocus}
      className={classes.textField}
      value={value}
      inputRef={ref}
      InputProps={{
        classes: {
          input: classes.input,
        },
        ...other,
      }}
    />
  )
}

function renderSuggestion(suggestion, { query, isHighlighted }) {
  const matches = match(suggestion.label, query)
  const parts = parse(suggestion.label, matches)

  return (
    <MenuItem selected={isHighlighted} component="div">
      <div>
        {parts.map((part, index) => {
          return part.highlight ? (
            <span key={index} style={{ fontWeight: 300 }}>
              {part.text}
            </span>
          ) : (
              <strong key={index} style={{ fontWeight: 500 }}>
                {part.text}
              </strong>
            )
        })}
      </div>
    </MenuItem>
  )
}

function renderSuggestionsContainer(options) {
  const { containerProps, children } = options

  return (
    <Paper {...containerProps} square>
      {children}
    </Paper>
  )
}

class CategorySelect extends Component {
  constructor() {
    super()

    this.state = {
      value: '',
      suggestions: []
    }

    this.handleQueryInput = this.handleQueryInput.bind(this)
    this.handleQuerySelect = this.handleQuerySelect.bind(this)
  }


  /** events */

  handleSuggestionsFetchRequested = ({ value }) => {
    this.setState({
      suggestions: this.getSuggestions(value)
    })
  }

  handleSuggestionsClearRequested = () => {
    this.setState({
      suggestions: []
    })
  }

  handleQueryInput(event, { newValue }) {
    this.setState({
      value: newValue
    })
  }

  handleQuerySelect(value) {
    if (value === '') { return }
    this.setState({ value })
    this.props.openResultsModal(value)
  }

  // handleQuerySelect = (value) => {
  //   if (value === '') { return }
  //   this.setState({ value })
  //   this.props.openResultsModal(value)
  // }

  getSuggestionValue = (suggestion) => {
    this.handleQuerySelect(suggestion.label)

    return suggestion.label
  }

  getSuggestions = (value) => {
    const inputValue = value.trim().toLowerCase()
    const inputLength = inputValue.length
    let count = 0

    return inputLength === 0
      ? []
      : this.props.categories.filter(suggestion => {
        const keep =
          count < 5 && suggestion.label.toLowerCase().slice(0, inputLength) === inputValue;

        if (keep) {
          count += 1;
        }

        return keep;
      })
  }

  render() {
    const { classes } = this.props

    return (
      <Autosuggest
        theme={{
          container: classes.container,
          suggestionsContainerOpen: classes.suggestionsContainerOpen,
          suggestionsList: classes.suggestionsList,
          suggestion: classes.suggestion,
        }}
        renderInputComponent={renderInput}
        suggestions={this.state.suggestions}
        onSuggestionsFetchRequested={this.handleSuggestionsFetchRequested}
        onSuggestionsClearRequested={this.handleSuggestionsClearRequested}
        renderSuggestionsContainer={renderSuggestionsContainer}
        getSuggestionValue={this.getSuggestionValue}
        renderSuggestion={renderSuggestion}
        inputProps={{
          autoFocus: true,
          classes,
          placeholder: 'Select a Category...',
          value: this.state.value,
          onChange: this.handleQueryInput
        }}
      />
    )
  }
}

export default withStyles(styles)(CategorySelect)
