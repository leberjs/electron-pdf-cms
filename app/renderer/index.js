import React from 'react'
import { render } from 'react-dom'
import { MuiThemeProvider, createMuiTheme } from 'material-ui/styles'

import Root from './containers/Root'

require('../static/fonts/roboto.scss')
require('../static/app.scss')

const theme = createMuiTheme({
  palette: {
    type: 'dark'
  }
})

render(
  <MuiThemeProvider theme={theme}>
    <Root />
  </MuiThemeProvider>,
  document.querySelector('#root')
)