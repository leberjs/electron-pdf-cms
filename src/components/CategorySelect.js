import React, { Component } from 'react'
import SelectField from 'material-ui/SelectField'
import MenuItem from 'material-ui/MenuItem'

class CategorySelect extends Component {
  constructor() {
    super()

    this.state = {
      value: 1
    }

    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(event, index, value) {
    this.setState({ value })
  }

  render() {
    return (
      <div className="container">
        <h2>Search By Category</h2>
        <SelectField
          floatingLabelText="Choose a Category"
          value={this.state.value}
          onChange={this.handleChange}
        >
          <MenuItem value={1} primaryText="Category A" />
          <MenuItem value={2} primaryText="Category B" />
          <MenuItem value={3} primaryText="Category C" />
        </SelectField>
      </div>
    )
  }
}

export default CategorySelect