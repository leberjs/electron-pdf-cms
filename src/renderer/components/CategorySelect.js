import React, { Component } from 'react'
import { withStyles } from 'material-ui/styles';
import { FormControl, FormHelperText } from 'material-ui/Form'
import Input, { InputLabel } from 'material-ui/Input'
import Select from 'material-ui/Select'
import { MenuItem } from 'material-ui/Menu'

import { getDirectories } from '../../main/directoryUtil'

const styles = theme => ({
  container: {
    margin: {
      top: 45,
      bottom: 50
    }
  },
  formControl: {
    //margin: theme.spacing.unit,
    minWidth: 300,
  }
})

const ITEM_HEIGHT = 48;
const ITEM_PADDING_TOP = 8;

class CategorySelect extends Component {
  constructor() {
    super()

    this.state = {
      categories: [],
      value: ''
    }

    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(e) {
    this.setState({ value: e.target.value })
  }

  componentDidMount() {
    let dirs = getDirectories()
    console.log(dirs)
    this.setState({ categories: dirs })
  }

  render() {
    const { classes } = this.props

    return (
      <div className={classes.container}>
        <h2>Search By Category</h2>
        <form autoComplete="off">
        <FormControl className={classes.formControl}>
          <InputLabel htmlFor="category-select">Select a Category...</InputLabel>
          <Select
            value={this.state.value}
            onChange={this.handleChange}
            input={<Input id="category-select" />}
            MenuProps={{
              PaperProps: {
                style: {
                  maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
                  width: 200,
                },
              },
            }}
          >
            <MenuItem value="">
              <em>None</em>
            </MenuItem>
            {this.state.categories.map((val, i) => <MenuItem key={i} value={i + 1}>{val}</MenuItem>)}
          </Select>
        </FormControl>
        </form>
      </div>
    )
  }
}

export default withStyles(styles)(CategorySelect)