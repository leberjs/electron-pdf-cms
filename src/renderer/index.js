import React from 'react'
import { render } from 'react-dom'
import { MuiThemeProvider, createMuiTheme } from 'material-ui/styles'

import App from './containers/App'

require('../static/fonts/roboto.scss')
require('../static/app.scss')

const theme = createMuiTheme({
  palette: {
    type: 'dark'
  }
})

const Root = () => (
  <MuiThemeProvider theme={theme}>
    <App />
  </MuiThemeProvider>
)

render(<Root />, document.querySelector('#root'))