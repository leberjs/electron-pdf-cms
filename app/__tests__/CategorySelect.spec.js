import React from 'react'
import { shallow, mount, render } from 'enzyme'
import { createShallow } from 'material-ui/test-utils'
import renderer from 'react-test-renderer'

import CategorySelect from '../renderer/components/CategorySelect'

describe('Component: CategorySelect', () => {
  let shallow

  beforeEach(() => {
    shallow = createShallow()
  })

  it('should render without crashing', () => {
    const wrapper = shallow(<CategorySelect />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have no category selected when first rendered', () => {
    const wrapper = mount(<CategorySelect />)
    const categoryInput = wrapper.find('TextField')

    expect(categoryInput.props().value).toBe('')
    expect(wrapper).toMatchSnapshot()
  })

  // it('should update state "value" when a select change is made', () => {
  //   const wrapper = mount(<CategorySelect />)
  //   const select = wrapper.find('SelectField')

  //   const onChange = jest.fn()

  //   select.simulate('change')
  //   expect(onChange).toHaveBeenCalled()
  // })
})
