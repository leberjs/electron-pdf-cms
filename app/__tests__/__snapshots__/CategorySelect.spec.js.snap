// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CategorySelect should have no category selected when first rendered 1`] = `
<withStyles(CategorySelect)>
  <CategorySelect
    classes={
      Object {
        "container": "CategorySelect-container-1",
        "suggestion": "CategorySelect-suggestion-3",
        "suggestionsContainerOpen": "CategorySelect-suggestionsContainerOpen-2",
        "suggestionsList": "CategorySelect-suggestionsList-4",
        "textField": "CategorySelect-textField-5",
      }
    }
  >
    <Autosuggest
      alwaysRenderSuggestions={false}
      focusInputOnSuggestionClick={true}
      getSuggestionValue={[Function]}
      highlightFirstSuggestion={false}
      id="1"
      inputProps={
        Object {
          "autoFocus": true,
          "classes": Object {
            "container": "CategorySelect-container-1",
            "suggestion": "CategorySelect-suggestion-3",
            "suggestionsContainerOpen": "CategorySelect-suggestionsContainerOpen-2",
            "suggestionsList": "CategorySelect-suggestionsList-4",
            "textField": "CategorySelect-textField-5",
          },
          "onChange": [Function],
          "placeholder": "Select a Category...",
          "value": "",
        }
      }
      multiSection={false}
      onSuggestionsClearRequested={[Function]}
      onSuggestionsFetchRequested={[Function]}
      renderInputComponent={[Function]}
      renderSuggestion={[Function]}
      renderSuggestionsContainer={[Function]}
      shouldRenderSuggestions={[Function]}
      suggestions={Array []}
      theme={
        Object {
          "container": "CategorySelect-container-1",
          "suggestion": "CategorySelect-suggestion-3",
          "suggestionsContainerOpen": "CategorySelect-suggestionsContainerOpen-2",
          "suggestionsList": "CategorySelect-suggestionsList-4",
        }
      }
    >
      <Autowhatever
        getSectionItems={[Function]}
        highlightedItemIndex={null}
        highlightedSectionIndex={null}
        id="1"
        inputProps={
          Object {
            "autoFocus": true,
            "classes": Object {
              "container": "CategorySelect-container-1",
              "suggestion": "CategorySelect-suggestion-3",
              "suggestionsContainerOpen": "CategorySelect-suggestionsContainerOpen-2",
              "suggestionsList": "CategorySelect-suggestionsList-4",
              "textField": "CategorySelect-textField-5",
            },
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "onKeyDown": [Function],
            "placeholder": "Select a Category...",
            "value": "",
          }
        }
        itemProps={[Function]}
        items={Array []}
        multiSection={false}
        renderInputComponent={[Function]}
        renderItem={[Function]}
        renderItemData={
          Object {
            "query": "",
          }
        }
        renderItemsContainer={[Function]}
        renderSectionTitle={[Function]}
        theme={
          Object {
            "container": "CategorySelect-container-1",
            "item": "CategorySelect-suggestion-3",
            "itemsContainerOpen": "CategorySelect-suggestionsContainerOpen-2",
            "itemsList": "CategorySelect-suggestionsList-4",
          }
        }
      >
        <div
          className="CategorySelect-container-1"
          key="react-autowhatever-1-container"
        >
          <TextField
            InputProps={
              Object {
                "aria-activedescendant": null,
                "aria-autocomplete": "list",
                "aria-expanded": false,
                "aria-haspopup": false,
                "aria-owns": "react-autowhatever-1",
                "autoComplete": "off",
                "classes": Object {
                  "input": undefined,
                },
                "key": "react-autowhatever-1-input",
                "onBlur": [Function],
                "onChange": [Function],
                "onFocus": [Function],
                "onKeyDown": [Function],
                "placeholder": "Select a Category...",
                "role": "combobox",
                "style": Object {},
                "type": "text",
              }
            }
            autoFocus={true}
            className="CategorySelect-textField-5"
            inputRef={[Function]}
            required={false}
            select={false}
            value=""
          >
            <withStyles(FormControl)
              className="CategorySelect-textField-5"
              required={false}
            >
              <FormControl
                className="CategorySelect-textField-5"
                classes={
                  Object {
                    "fullWidth": "MuiFormControl-fullWidth-9",
                    "marginDense": "MuiFormControl-marginDense-8",
                    "marginNormal": "MuiFormControl-marginNormal-7",
                    "root": "MuiFormControl-root-6",
                  }
                }
                component="div"
                disabled={false}
                error={false}
                fullWidth={false}
                margin="none"
                required={false}
              >
                <div
                  className="MuiFormControl-root-6 CategorySelect-textField-5"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  required={false}
                >
                  <withStyles(Input)
                    aria-activedescendant={null}
                    aria-autocomplete="list"
                    aria-expanded={false}
                    aria-haspopup={false}
                    aria-owns="react-autowhatever-1"
                    autoComplete="off"
                    autoFocus={true}
                    classes={
                      Object {
                        "input": undefined,
                      }
                    }
                    inputRef={[Function]}
                    key="react-autowhatever-1-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Select a Category..."
                    role="combobox"
                    style={Object {}}
                    type="text"
                    value=""
                  >
                    <Input
                      aria-activedescendant={null}
                      aria-autocomplete="list"
                      aria-expanded={false}
                      aria-haspopup={false}
                      aria-owns="react-autowhatever-1"
                      autoComplete="off"
                      autoFocus={true}
                      classes={
                        Object {
                          ".MuiInput-error-13:after": "MuiInput-error-13:after",
                          ".MuiInput-inkbar-12.MuiInput-focused-18:after": "MuiInput-inkbar-12.MuiInput-focused-18:after",
                          ".MuiInput-inkbar-12:after": "MuiInput-inkbar-12:after",
                          ".MuiInput-input-14:-ms-input-placeholder": "MuiInput-input-14:-ms-input-placeholder",
                          ".MuiInput-input-14::-moz-placeholder": "MuiInput-input-14::-moz-placeholder",
                          ".MuiInput-input-14::-ms-input-placeholder": "MuiInput-input-14::-ms-input-placeholder",
                          ".MuiInput-input-14::-webkit-input-placeholder": "MuiInput-input-14::-webkit-input-placeholder",
                          ".MuiInput-input-14::-webkit-search-decoration": "MuiInput-input-14::-webkit-search-decoration",
                          ".MuiInput-input-14:focus": "MuiInput-input-14:focus",
                          ".MuiInput-input-14:invalid": "MuiInput-input-14:invalid",
                          ".MuiInput-underline-19.MuiInput-disabled-17:before": "MuiInput-underline-19.MuiInput-disabled-17:before",
                          ".MuiInput-underline-19:before": "MuiInput-underline-19:before",
                          ".MuiInput-underline-19:hover:not(.MuiInput-disabled-17):before": "MuiInput-underline-19:hover:not(.MuiInput-disabled-17):before",
                          "disabled": "MuiInput-disabled-17",
                          "error": "MuiInput-error-13",
                          "focused": "MuiInput-focused-18",
                          "formControl": "MuiInput-formControl-11",
                          "fullWidth": "MuiInput-fullWidth-25",
                          "inkbar": "MuiInput-inkbar-12",
                          "input": "MuiInput-input-14",
                          "inputAdorned": "MuiInput-inputAdorned-15",
                          "inputDense": "MuiInput-inputDense-16",
                          "inputDisabled": "MuiInput-inputDisabled-21",
                          "inputMultiline": "MuiInput-inputMultiline-24",
                          "inputSearch": "MuiInput-inputSearch-23",
                          "inputSingleline": "MuiInput-inputSingleline-22",
                          "label + .MuiInput-formControl-11": "abel + .MuiInput-formControl-11",
                          "label[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14": "abel[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14",
                          "label[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14:-ms-input-placeholder": "abel[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14:-ms-input-placeholder",
                          "label[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14::-moz-placeholder": "abel[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14::-moz-placeholder",
                          "label[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14::-ms-input-placeholder": "abel[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14::-ms-input-placeholder",
                          "label[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14::-webkit-input-placeholder": "abel[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14::-webkit-input-placeholder",
                          "label[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14:focus:-ms-input-placeholder": "abel[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14:focus:-ms-input-placeholder",
                          "label[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14:focus::-moz-placeholder": "abel[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14:focus::-moz-placeholder",
                          "label[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14:focus::-ms-input-placeholder": "abel[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14:focus::-ms-input-placeholder",
                          "label[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14:focus::-webkit-input-placeholder": "abel[data-shrink=false] + .MuiInput-formControl-11 .MuiInput-input-14:focus::-webkit-input-placeholder",
                          "multiline": "MuiInput-multiline-20",
                          "root": "MuiInput-root-10",
                          "underline": "MuiInput-underline-19",
                        }
                      }
                      disableUnderline={false}
                      fullWidth={false}
                      inputRef={[Function]}
                      multiline={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Select a Category..."
                      role="combobox"
                      style={Object {}}
                      type="text"
                      value=""
                    >
                      <div
                        aria-activedescendant={null}
                        aria-autocomplete="list"
                        aria-expanded={false}
                        aria-haspopup={false}
                        aria-owns="react-autowhatever-1"
                        className="MuiInput-root-10 MuiInput-formControl-11 MuiInput-inkbar-12 MuiInput-underline-19"
                        onBlur={[Function]}
                        onFocus={[Function]}
                        role="combobox"
                        style={Object {}}
                      >
                        <input
                          autoComplete="off"
                          autoFocus={true}
                          className="MuiInput-input-14 MuiInput-inputSingleline-22"
                          disabled={false}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          placeholder="Select a Category..."
                          type="text"
                          value=""
                        />
                      </div>
                    </Input>
                  </withStyles(Input)>
                </div>
              </FormControl>
            </withStyles(FormControl)>
          </TextField>
          <withStyles(Paper)
            id="react-autowhatever-1"
            key="react-autowhatever-1-items-container"
            square={true}
            style={Object {}}
          >
            <Paper
              classes={
                Object {
                  "root": "MuiPaper-root-26",
                  "rounded": "MuiPaper-rounded-27",
                  "shadow0": "MuiPaper-shadow0-28",
                  "shadow1": "MuiPaper-shadow1-29",
                  "shadow10": "MuiPaper-shadow10-38",
                  "shadow11": "MuiPaper-shadow11-39",
                  "shadow12": "MuiPaper-shadow12-40",
                  "shadow13": "MuiPaper-shadow13-41",
                  "shadow14": "MuiPaper-shadow14-42",
                  "shadow15": "MuiPaper-shadow15-43",
                  "shadow16": "MuiPaper-shadow16-44",
                  "shadow17": "MuiPaper-shadow17-45",
                  "shadow18": "MuiPaper-shadow18-46",
                  "shadow19": "MuiPaper-shadow19-47",
                  "shadow2": "MuiPaper-shadow2-30",
                  "shadow20": "MuiPaper-shadow20-48",
                  "shadow21": "MuiPaper-shadow21-49",
                  "shadow22": "MuiPaper-shadow22-50",
                  "shadow23": "MuiPaper-shadow23-51",
                  "shadow24": "MuiPaper-shadow24-52",
                  "shadow3": "MuiPaper-shadow3-31",
                  "shadow4": "MuiPaper-shadow4-32",
                  "shadow5": "MuiPaper-shadow5-33",
                  "shadow6": "MuiPaper-shadow6-34",
                  "shadow7": "MuiPaper-shadow7-35",
                  "shadow8": "MuiPaper-shadow8-36",
                  "shadow9": "MuiPaper-shadow9-37",
                }
              }
              component="div"
              elevation={2}
              id="react-autowhatever-1"
              square={true}
              style={Object {}}
            >
              <div
                className="MuiPaper-root-26 MuiPaper-shadow2-30"
                id="react-autowhatever-1"
                style={Object {}}
              />
            </Paper>
          </withStyles(Paper)>
        </div>
      </Autowhatever>
    </Autosuggest>
  </CategorySelect>
</withStyles(CategorySelect)>
`;

exports[`Component: CategorySelect should render without crashing 1`] = `
<CategorySelect
  classes={
    Object {
      "container": "CategorySelect-container-1",
      "suggestion": "CategorySelect-suggestion-3",
      "suggestionsContainerOpen": "CategorySelect-suggestionsContainerOpen-2",
      "suggestionsList": "CategorySelect-suggestionsList-4",
      "textField": "CategorySelect-textField-5",
    }
  }
/>
`;
